src/Compilers/Core/Portable/CommandLine/CommonCompiler.ExistingReferencesResolver.cs

```diff
@@ -73,13 +72,15 @@ public override int GetHashCode()
                 return _resolver.GetHashCode();
             }
 
-            public bool Equals(ExistingReferencesResolver other)
+            public bool Equals(ExistingReferencesResolver? other)
             {
-                return _resolver.Equals(other._resolver) &&
-                       _availableReferences.SequenceEqual(other._availableReferences);
+                return
+                    other is object &&
+                    _resolver.Equals(other._resolver) &&
+                    _availableReferences.SequenceEqual(other._availableReferences);
             }
 
-            public override bool Equals(object other) => Equals(other as ExistingReferencesResolver);
```

Comment:
>Equals(other as ExistingReferencesResolver) [](start = 57, length = 43)

Consider leaving the implementation unchanged since the other `Equals()` overload already checks `null`. Same comment for other `Equals(object)` implementations.

Reply:
The `as` operator doesn't support `<type name>?` hence I had to either issue a suppression here or rewrite. Decided on the whole it was better to make this have the idiomatic `Equals` the rest of the code base has.

Reply:
`as` supports `as <type name>`.

---
In reply to: [512992047](https://github.com/dotnet/roslyn/pull/48930#discussion_r512992047) [](ancestors = 512992047)